Lib/ntpath.py

```diff
@@ -267,6 +267,24 @@ def islink(path):
         return False
     return stat.S_ISLNK(st.st_mode)
 
+
+# Is a path a junction?
+
+if hasattr(os.stat_result, 'st_reparse_tag'):
+    def isjunction(path):
+        """Test whether a path is a junction"""
+        try:
+            st = os.lstat(path)
+        except (OSError, ValueError, AttributeError):
+            return False
+        return bool(st.st_reparse_tag & stat.IO_REPARSE_TAG_MOUNT_POINT)
```

Comment:
Sorry, using a bitwise AND here was an editing mistake in my original message on python-ideas. This needs to be an equality check:

```suggestion
        return bool(st.st_reparse_tag == stat.IO_REPARSE_TAG_MOUNT_POINT)
```
---

The upper 16 bits of a reparse tag are flags, of which 12 bits are reserved, and the lower 16 bits are the assigned tag number in one of two domains: Microsoft and non-Microsoft. For example, `IO_REPARSE_TAG_MOUNT_POINT` (0xA000_0003) and `IO_REPARSE_TAG_SYMLINK` (0xA000_000C) have the flags set to indicate a Microsoft tag (0x8000_0000) and name-surrogate type (0x2000_0000). Currently there are about 50 Microsoft tags and about 80 non-Microsoft tags.